webapp/src/ts/components/filters/analytics-filter/analytics-filter.component.ts

```diff
@@ -30,9 +65,67 @@ export class AnalyticsFilterComponent implements AfterContentInit, AfterContentC
     this.subscriptions.unsubscribe();
   }
 
+  private subscribeToStore() {
+    const subscription = this.store.select(Selectors.getSidebarFilter).subscribe(({ isOpen }) => {
+      this.isOpen = isOpen;
+    });
+    this.subscriptions.add(subscription);
+  }
+
   private setActiveModule() {
     this.activeModule = this.analyticsModules?.find(module => {
       return module.id === this.route.snapshot?.firstChild?.data?.moduleId;
     });
   }
+
+  private subscribeToRouteChanges() {
+    const routeSubscription = this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd)
+    ).subscribe(() => {
+      this.canDisplayFilterButton();
+    });
+    this.subscriptions.add(routeSubscription);
+  }
+
+  private hasMultipleFacilities() {
+    return this.userSettingsService
+      .get()
+      .then((userSettings: any) => {
+        const facility = userSettings.facility_id;
+        return Array.isArray(facility) && facility.length > 1;
+      });
+  }
+
+  private checkPermissions() {
+    return Promise
+      .all([
+        this.authService.has('!can_view_old_filter_and_search'),
+        this.authService.has('!can_view_old_action_bar')
+      ])
+      .then(([cannotViewOldFilter, cannotViewOldActionBar]) => {
+        return cannotViewOldFilter && cannotViewOldActionBar;
+      })
```

Comment:
The auth service supports an array for checking multiple permissions, so it doesn't need 2 separate calls.